### ISSUE TITLE
Suggest using operator instead of explicit operator method call

### ISSUE THREAD

**Issue by Robzz** [Manishearth/rust-clippy#217]
This lint would trigger on something like
```rust
let i: i32 = 5;
let b = i.add(3);
```
And suggest something along the lines of `this expression can be simplified to 'i + 3'` This makes the code more concise and IMHO a bit clearer. This should probably be `Allow` by default.

Thoughts?

**Comment by birkenfeld** [comment]
For me this is in "whoever writes that must have reasons for it" territory. (Maybe to make it easier to switch to `saturating_add` or `wrapping_add` later...) But it's (hopefully) not a newbie thing...

**Comment by llogiq** [comment]
Also the lint should ensure that the `std::ops::Add::add(..)` function is actually the one that's called. Other types could conceivably have their own `add(..)` method without implementing `Add`.

**Comment by birkenfeld** [comment]
To make it clear, my previous comment was supposed to suggest not to implement such a lint at all...

**Comment by llogiq** [comment]
Agreed, but even if one were to implement it, they'd have to guard against false positives.

**Comment by Robzz** [comment]
Well, I was asking because I couldn't decide between "might possibly be useful" and "meh", so I guess you both answered that. I'll close unless I come up with a good reason to do this. But in hindsight, I could have known this is of limited use.
